client/src/partials/patients/groups/groups.js

```diff
@@ -1,127 +1,126 @@
+/** declaring the controller to the bhima module **/
 angular.module('bhima.controllers')
-.controller('patientGroup', [
-  '$scope',
-  '$translate',
-  '$window',
-  '$location',
-  'connect',
-  'validate',
-  'appstate',
-  'messenger',
-  'uuid',
-  function ($scope, $translate, $window, $location, connect, validate, appstate, messenger, uuid) {
-    var dependencies = {};
-    var session = $scope.session = {
-      selected : null
-    };
-
-    dependencies.group = {
-
-       query : {
-        identifier : 'uuid',
-        tables : {
-          'patient_group' : {columns : ['uuid', 'name', 'price_list_uuid', 'subsidy_uuid']}
-        }
-      }
-    };
-
-    dependencies.subsidies = {
-      query : {
-        identifier : 'uuid',
-        tables : {
-          'subsidy' : {columns : ['uuid', 'text']}
-        }
-      }
-    };
-
-    dependencies.list = {
-      query : {
-        identifier : 'uuid',
-        tables : {
-          'price_list' : {
-            columns : ['uuid', 'title']
-          }
-        }
-      }
-    };
-
-    appstate.register('enterprise', loadEnterprise);
-
-    function loadEnterprise(enterprise) {
-      $scope.enterprise = enterprise;
-      dependencies.group.query.where = ['patient_group.enterprise_id='+enterprise.id];
-      validate.process(dependencies).then(initialisePatientGroup);
-    }
+.controller('patientGroupController', patientGroupController);
 
-    function initialisePatientGroup(model) {
-      angular.extend($scope, model);
-    }
+/** inject service(s) or dependencies for the modules **/
+patientGroupController.$inject = ['PatientGroupService', 'PriceListService', 'SessionService', '$window'];
 
-    $scope.remove = function (grp) {
-      if (!$window.confirm($translate.instant('PATIENT_GRP.CONFIRM_MESSAGE'))) {
-        return;
-      }
+function patientGroupController (patientGroupService, priceListService, sessionService, $window){
+  var vm = this; 
 
-      connect.delete('patient_group', 'uuid', [grp.uuid])
-      .then(function () {
-        $scope.group.remove(grp.uuid);
-      }, function (err) {
-        messenger.danger('error:' + JSON.stringify(err));
-      });
-    };
-
-    // register namespace
-
-    $scope.newGroup = function () {
-      $scope.register = {};
-      $scope.action = 'register';
-      session.selected = null;
-    };
-
-    $scope.saveRegistration = function () {
-      var packaged = connect.clean($scope.register);
-      packaged.uuid = uuid();
-      packaged.enterprise_id = $scope.enterprise.id;
-
-      // validate that register is complete.
-      connect.post('patient_group', packaged)
-      .then(function () {
-        $scope.group.post(packaged);
-        $scope.edit(packaged);
-      });
-    };
+  /** This method is responsible of initializing data **/
+  function startup (){
+
+    /** make the loading state into true, while loading data **/
+    vm.loading = true;
+
+    /** fetching all price list **/
```

Comment:
This is a very minor (non blocking) review on comments but single line comments like this can be formatted as 
```js 
// fetching all prices lists 
``` 

There are some exceptions with JSDoc that can be discovered by reading its documentation (for example documenting a single line exposed method in this style)
```js
/** This method is responsible for retrieving all price lists from the server /price_list API. */
service.priceLists = priceList;
```